Exercise04/MainActivity.cs

```diff
@@ -0,0 +1,198 @@
+
+                    break;
+                case "*":
+                    recentValue *= currentValue;
+
+                    break;
+                case "/":
+                    recentValue /= currentValue;
+
+                    break;
+                case "%":
+                    recentValue = currentValue / 100;
+                    currentOperator = "";
+
+                    break;
+                case "":
+                    check = false;
+                    recentValue = currentValue;
+
+                    break;
+            }
+
+            if (check)
+            {
+                result.SetText(recentValue.ToString(), TextView.BufferType.Editable);
+            }
+        }
+
+        public void SetCurrentValue(string str)
+        {
+            result.SetText(str, TextView.BufferType.Editable);
+        }
+
+        public void SetValueDefault()
+        {
+            SetCurrentValue("0");
+        }
+
+        public void Operate(string operate)
+        {
+            currentValue = double.Parse(result.Text);
+
+            if (operate.Equals("c"))
+            {
+                SetValueDefault();
+                recentValue = 0;
+                currentValue = 0;
+                currentOperator = "";
+
+                return;
+            }
+
+            if (operate.Equals("d"))
+            {
+                if (result.Text.Length > 1)
+                {
+                    SetCurrentValue(result.Text.Remove(result.Text.Length - 1));
+                }
+                else
+                {
+                    SetValueDefault();
+                }
+
+                return;
+            }
+
+            switch (operate)
+            {
+                case "+":
+                    Calculate();
+                    currentOperator = "+";
+
+                    break;
+                case "-":
+                    Calculate();
+                    currentOperator = "-";
+
+                    break;
+                case "*":
+                    Calculate();
+                    currentOperator = "*";
+
+                    break;
+                case "/":
+                    Calculate();
+                    currentOperator = "/";
+
+                    break;
+                case "%":
+                    currentOperator = "%";
+                    Calculate();
+
+                    break;
+                case "=":
+                    Calculate();
+                    currentOperator = "";
+
+                    break;
+            }
+
+            reset = true;
+        }
+
+        public void Addvalue(string str)
+        {
+            if (reset)
+            {
+                SetValueDefault();
+                reset = false;
+            }
+
+            if (result.Text.Equals("0")) 
+            {
+                if (str.Equals("."))
+                {
+                    result.SetText(result.Text + str, TextView.BufferType.Editable);

```

Comment:
Use the method you have already declared `SetCurrentValue`

Reply:
Done